Задания с 30.11.2021/30.11.2021-Task1/30.11.2021-Task1/SubstringSearch.c

```diff
@@ -0,0 +1,152 @@
+﻿#pragma warning (disable: 4668 4996 5045 6031)
+
+#define MAX_STRING_LENGTH 100
+#define MAX_TEXT_LENGTH 10000
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include <string.h>
+#include <Windows.h>
+
+int hash(const char string[], const int module)
+{
+    const int length = (int)strlen(string);
+    const int primeNumber = 3;
+    int hash = 0;
+
+    for (int i = 0; i < length; i++)
+    {
+        hash = (hash * primeNumber + (unsigned char)string[i]) % module;
+    }
+
+    return hash;
+}
+
+void getStringFragment(const char source[], char destination[], const int beginPosition, const int endPosition)
+{
+    for (int i = 0; i < endPosition - beginPosition + 1; i++)
+    {
+        destination[i] = source[beginPosition + i];
+    }
+}
+
+int RabinKarpSearch(const char string[], const char substring[])
```

Comment:
```suggestion
int rabinKarpSearch(const char string[], const char substring[])
```
camelCase используется даже с фамилиями уважаемых учёных